src/internal/modules/contact-entry/plugin.js

```diff
@@ -1,10 +1,51 @@
-const routes = require('./routes');
+'use strict';
+const Joi = require('@hapi/joi');
+const routes = Object.values(require('./routes'));
+const session = require('./lib/session');
+
+const OPTIONS_SCHEMA = Joi.object({
+  back: Joi.string().required(),
+  caption: Joi.string().optional().default(null),
+  key: Joi.string().required(),
+  redirectPath: Joi.string().required(),
+  companyId: Joi.string().guid().required(),
+  data: Joi.object().optional()
+});
+
+/**
+ * This function stores data in the session and returns
+ * a path which can start the flow
+ * @param {Object} options
+ * @return {String} path
+ */
+function contactEntryRedirect (options) {
+  // Validate options
+  Joi.assert(options, OPTIONS_SCHEMA);
+
+  // Store in session
+  session.set(this, options.key, options);
+
+  // Return redirect path to enter flow
+  return `/contact-entry/${options.key}/select-contact`;
+}
+
+/**
+ * Get the data set in the flow
+ * @param {String} key
+ * @return {Object}
+ */
+function getNewContact (key) {
+  return (session.get(this, key) || {}).data;
+}
 
 const contactEntryPlugin = {
-  register: (server) => {
-    for (const eachRoute of Object.values(routes)) {
-      server.route(eachRoute);
-    }
+  register: (server, options) => {
```

Comment:
do you need the options here as per sonar cloud?

Reply:
It is done :)